### ISSUE TITLE
Couldn't read values in CFPrefsPlistSource. CRASH

### ISSUE THREAD

**Issue by RadagastCH** [getsentry/sentry-cocoa#3801]
### Platform

macOS

### Environment

Production, Develop, TestFlight, Other

### Installed

Swift Package Manager

### Version

8.22.4

### Did it work on previous versions?

8.21.0

### Steps to Reproduce

1. macOS app with extensions that used Sentry
2. Try to run anything above 8.21.0 installed via SPM

### Expected Result

No Crash 

### Actual Result

![CleanShot 2024-03-27 at 12 03 33@2x](https://github.com/getsentry/sentry-cocoa/assets/34520744/7aa8478c-9683-485e-ae73-2079d7906306)

**Comment by philipphofmann** [comment]
@RadagastCH, are you 100% sure this error is related to the sentry-cocoa SDK? Does the error disappear when you roll back to `8.21.0`?

When looking at suggestions to solve this error on this [Stackoverflow post](https://stackoverflow.com/questions/38275395/failed-to-read-values-in-cfprefsplistsource-ios-10), I have a hard time figuring out how our SDK could cause this error.

**Comment by RadagastCH** [comment]
Yep, we don't have any problems with version 8.21.0, but we are experiencing crashes with any versions higher than that. We only change the version in SPM and make no other change.

**Comment by RadagastCH** [comment]
> @RadagastCH, are you 100% sure this error is related to the sentry-cocoa SDK? Does the error disappear when you roll back to `8.21.0`?
> 
> 
> 
> When looking at suggestions to solve this error on this [Stackoverflow post](https://stackoverflow.com/questions/38275395/failed-to-read-values-in-cfprefsplistsource-ios-10), I have a hard time figuring out how our SDK could cause this error.

Yep, we don't have any problems with version 8.21.0, but we are experiencing crashes with any versions higher than that. We only change the version in SPM and make no other change.

**Comment by philipphofmann** [comment]
@RadagastCH, again looking at this issue, I guess that it has something to do with 

> macOS app with extensions that used Sentry

What exactly do you mean by that? Which extensions are you using?

**Comment by RadagastCH** [comment]
@philipphofmann 
It's more likely that the warning is not related to the problem. It seems the main issue is "Unable to find class: SentryCrashExceptionApplication, exiting."
![image](https://github.com/getsentry/sentry-cocoa/assets/34520744/397d266c-dd25-4ac1-95d5-78d7a8a89d47)

It looks like it's not a crash but just a termination.

In fact, we see these classes in the headers.

> @RadagastCH, again looking at this issue, I guess that it has something to do with
> 
> > macOS app with extensions that used Sentry
> 
> What exactly do you mean by that? Which extensions are you using?

We are using the app as an extension for Apple Photos and Adobe products.
But I'm not so sure it has anything to do with that.

**Comment by RadagastCH** [comment]
It appears we can resolve the issue by adding this code somewhere:

[SentryCrashExceptionApplication.sharedApplication description];

**Comment by kahest** [comment]
Thanks @RadagastCH - looks like this is related to some optimization/code stripping. As a potential workaround we could call `[SentryCrashExceptionApplication.sharedApplication description];` in the SDK code somewhere.

**Comment by philprime** [comment]
Closing this as outdated/superseded by other issues or newer SDK/OS versions. Feel free to comment with answers or updated information and we will reopen if it’s actionable.
